### ISSUE TITLE
int pack_image the 'subtract the black level and scale the data' :why is  im = np.maximum(im - 512, 0) / (16383 - 512)  

### ISSUE THREAD

**Issue by zhanglijian** [cchen156/Learning-to-See-in-the-Dark#35]
why black level equal 512 and why is 16383 ,What is meaning of the number 16383?

**Comment by cchen156** [comment]
The raw data is 14 bits. 2^14 = 16384. 

The black level is related to the hardware which can be tested.

**Comment by zhanglijian** [comment]
ok,I  have know that and I have test the Sony preTrained_modele on my on image, the result is good, but when I process the Bayer Data with the WhiteBlance ,the result is little bad than before.So the input Bayer Data should must be  the original raw DATA without any processes??

**Comment by cchen156** [comment]
Yes. If you do that by yourself, the data will be processed twice.

**Comment by yxlao** [comment]
FWIW, `512` is the black level, or, raw offset, which is a property of the camera as well. Offset `512` means value `512` is actually representing `0`.

It's possible to get the values the black level in `rawpy` as follows:
```python
import rawpy
im_raw = rawpy.imread("images/im_0.1s.arw")
print(im_raw.black_level_per_channel)
```

**Comment by pencilzhang** [comment]
> FWIW, `512` is the black level, or, raw offset, which is a property of the camera as well. Offset `512` means value `512` is actually representing `0`.
> 
> It's possible to get the values the black level in `rawpy` as follows:
> 
> ```python
> import rawpy
> im_raw = rawpy.imread("images/im_0.1s.arw")
> print(im_raw.black_level_per_channel)
> ```

Is this used to check variable black level offset per channel? 521 should be constant VIN BLc from the camera platform, no?
